ts/navigation/MainNavigator.tsx

```diff
@@ -139,30 +139,36 @@ const navigation = createBottomTabNavigator(
     }
   },
   {
-    navigationOptions: ({ navigation: nav }) => ({
+    defaultNavigationOptions: ({ navigation: nav }) => ({
       tabBarVisible: getTabBarVisibility(nav),
-      tabBarLabel: ({ tintColor }) => {
+      tabBarLabel: (options: {
+        tintColor: string | null;
+        focused: boolean;
+      }) => {
         const { routeName } = nav.state;
         // adding `color` as a separate style property since it depends on tintColor
         return (
           <Text
             style={[
               styles.labelStyle,
-              { color: tintColor === null ? undefined : tintColor }
+              {
+                color:
+                  options.tintColor === null ? undefined : options.tintColor
+              }
             ]}
           >
             {getLabel(routeName)}
           </Text>
         );
       },
-      tabBarIcon: ({ tintColor }) => {
+      tabBarIcon: (options: { tintColor: string | null; focused: boolean }) => {
```

Comment:
TS doesn't infer the type of the function here?

Reply:
Typescript infer "any" so i have to specify the exact type